tools/depcheck/README.md

```diff
@@ -0,0 +1,176 @@
+depcheck
+========
+
+Dependency analysis tool for Go repositories.
+
+### Why
+
+In the case of OpenShift, having a flat `vendor` dependency tree that brings in a
+large enough dependency, such as `kubernetes`, creates a problem with conflicting
+dependencies. It may be the case that we depend on a specific version of package `A`,
+while `kubernetes` (or one of its dependencies) depends on that same package `A`, 
+but a different version of it.
+Because only one version of package `A` may exist under the `vendor` directory, it
+might be useful to know:
+  - Which parts of our codebase depend on package `A`, and are responsible for the conflict?
+  - Would those parts of our codebase work with the version of package `A` that is brought
+  in by `kubernetes` (or one of its dependencies)?
+  - If not, can we break our dependency on package `A`?
+ 
+Both the `analyze` and `trace` commands included in this tool are meant to aid in 
+answering the above questions.
+
+### Getting started
+
+#### Limitations
+
+Before beginning to use this tool, it is important to note that because both the `trace`
+and `analyze` commands make use of [go list](https://golang.org/cmd/go/#hdr-List_packages)
+in order to obtain information about packages in a Go repository, there are limitations
+that exist for these two commands:  
+
+  - No symlinked directories will be traversed in a given repo. Any resulting graph information generated
```

Comment:
Since we heavily rely on this I'd like to see a copy&paste command how to solve this for origin.

Reply:
Ok, since I'm commenting as I'm reading I've got to a place where you do so, can you link it here?

Reply:
Sure, will link Origin example here

Reply:
Done